### PR TITLE
Update submodules & Fix S3 Tests

### PR DESCRIPTION
**Pull request by waahm7**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/test_s3.py [hunk 1]
```diff
@@ -240,6 +242,7 @@ def test_put_object_multiple_times(self):
                 on_headers=self._on_request_headers,
                 on_body=self._on_request_body)
             finished_futures.append(s3_request.finished_future)
+            del s3_request
```
Comment (TingDaoK) [inline test/test_s3.py:245]: why this is not released with as it goes out of scope?

Comment (graebm) [inline test/test_s3.py:245]:
```suggestion
            # request keeps connection alive. delete pointer so connection can shut down
            del s3_request
```

**Review by graebm — APPROVED** [review]
fix & ship

File: test/test_s3.py (hunk 1 above)
Comment (waahm7) [reply test/test_s3.py:245]: I had a discussion with Mike and Python has a weird behavior where it is not guaranteed that something will delete when it goes out of scope.

**Pull request merged**